pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -161,3 +180,73 @@ func detectCPUManagerFile(cpuManagerPaths []string) (string, error) {
 	}
 	return "", fmt.Errorf("no cpumanager policy file found")
 }
+
+func isVirtualizationSupportedOnNodeImpl() bool {
+	// If an error occurs in this process (can't parse info file / any other unexpected failure) this function will
+	// return true just to stay on the safe side. false will be returned only if we are certain that virtualization
+	// is not supported.
+
+	const cpuInfoFilePath = "/proc/cpuinfo"
+
+	cpuInfoFile, err := os.Open(cpuInfoFilePath)
+	if err != nil {
+		log.Log.Reason(err).Errorf("cannot open %s", cpuInfoFilePath)
+		return true
+	}
+	defer cpuInfoFile.Close()
+
+	scanner := bufio.NewScanner(cpuInfoFile)
+	for scanner.Scan() {
+		line := scanner.Bytes()
+		if bytes.Contains(line, []byte("vmx")) || bytes.Contains(line, []byte("svm")) {
+			return true
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Log.Reason(err).Errorf("error while parsing %s", cpuInfoFilePath)
+		return true
+	}
+
+	return false
+}
+
+func (h *HeartBeat) setSchedulable(labels labelType, annotations annotationType) {
+	isKubevirtSchedulable := true
+
+	if !h.deviceManagerController.Initialized() {
+		isKubevirtSchedulable = false
+	}
+
+	if isKubevirtSchedulable && !h.clusterConfig.AllowEmulation() {
+		// Host's virtualization capabilities are assumed to never change. Therefore use cached result if exists.
```

Comment:
As it is right now, we are determining kvm-availability by checking for the file. So basically on the module level. There it is also possible to unlload the kvm module and our device plugin would react. You would probably also want to see it non-schedulable in that case, but I am not sure if we should pursue this, since the device plugins already handle all that.